Sources/MapboxCoreNavigation/EHorizon/RoadObject.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+import CoreLocation
+import MapboxNavigationNative
+
+/**
+ * Describes the object on the road.
+ * There are two sources of road objects: active route and the electronic horizon.
+ */
+public struct RoadObject {
+
+    /**
+     * Id of the road object. If we get the same objects (e.g. `RoadObjectType.tunel`) from the
+     * electronic horizon and the active route, they will not have the same IDs.
+     */
+    public let identifier: RoadObjectIdentifier
+
+    /** Length of the object, `nil` if the object is point-like. */
+    public let length: CLLocationDistance?
+
+    /** Location of the road object. */
+    public let location: RoadObjectLocation
+
+    /** Type of the road object with metadata. */
+    public let type: RoadObjectType
+
+    /** `true` if an object is added by user, `false` if it comes from Mapbox service. */
+    public let isCustom: Bool
```

Comment:
Why do we need this?

Reply:
This is to distinguish road objects coming from Mapbox services and custom objects added by a user. 
More on custom objects in https://github.com/mapbox/mapbox-navigation-ios/pull/3004.